convert2rhel/actions/pre_ponr_changes/backup_system.py

```diff
@@ -63,3 +67,96 @@ def run(self):
 
         repo.backup_yum_repos()
         repo.backup_varsdir()
+
+
+class BackupPackageFiles(actions.Action):
+    id = "BACKUP_PACKAGE_FILES"
+
+    def run(self):
+        """Backup changed package files"""
+        loggerinst.task("Prepare: Backup package files")
+
+        super(BackupPackageFiles, self).run()
+
+        package_files_changes = self._get_rpm_va()
+
+        for file in package_files_changes:
+            if file["status"] == "missing":
+                pass  # no action needed now
+            elif re.search(r"[5]", file["status"]):
+                # If the MD5 checksum differs, the content of the file differs
+                restorable_file = backup.NewRestorableFile(file["path"])
+                backup.backup_control.push(restorable_file)
+
+    def _get_rpm_va(self):
+        """Get the output from rpm -Va command from generated during resolving system info
+        to get changes made to package files.
+        Return them as a list of dict, for example:
+        [{"status":"S5T", "file_type":"c", "path":"/etc/yum.repos.d/CentOS-Linux-AppStream.repo"}]
+        """
+        path = os.path.join(LOG_DIR, PRE_RPM_VA_LOG_FILENAME)
+        with open(path, "r") as f:
+            output = f.read()
+
+        return self._parse(output)
+
+    def _parse(self, input):
+        """Parse the output from input"""
+        input = input.strip()
+        lines = input.split("\n")
+        data = []
+
+        for line in lines:
+            parsed_line = self._parse_line(line.strip())
+            if parsed_line:
+                data.append(parsed_line)
+
+        return data
```

Comment:
What if we merge those two functions and have just one that do the parsing? 

```suggestion
    def _parse_rpm_va(self):
        """Get the output from rpm -Va command from generated during resolving system info
        to get changes made to package files.
        Return them as a list of dict, for example:
        [{"status":"S5T", "file_type":"c", "path":"/etc/yum.repos.d/CentOS-Linux-AppStream.repo"}]
        """
        path = os.path.join(LOG_DIR, PRE_RPM_VA_LOG_FILENAME)
        with open(path, "r") as f:
            output = f.read().strip()

        lines = output.split("\n")
        data = []

        for line in lines:
            parsed_line = self._parse_line(line.strip())
            if parsed_line:
                data.append(parsed_line)

        return data
```

Reply:
Yeah, it may be more efficient, we won't need to call the function and save a little of time